vmm/include/modules/template_tee.h

```diff
@@ -43,10 +43,12 @@ typedef struct {
 	uint8_t smc_param_to_ree[8];
 
 	/* If first SMC call is from REE, in that SMC, first_smc_to_tee() will be called.
-	   pre_world_switch()/post_world_switch() will only be called for other SMCs */
+	   pre_world_switch()/post_world_switch() will only be called for other SMCs.
+	   smc_hook() would be invoked during each world switch. */
 	void (*first_smc_to_tee)(guest_cpu_handle_t gcpu);
 	void (*pre_world_switch)(guest_cpu_handle_t gcpu);
 	void (*post_world_switch)(guest_cpu_handle_t gcpu, guest_cpu_handle_t gcpu_prev);
+	void (*smc_hook)(guest_cpu_handle_t cpu, guest_cpu_handle_t next_gcpu);
```

Comment:
don't introduce new API. you can use post_world_switch() here

Reply:
Done